doc/source/rllib/core-concepts.rst

```diff
@@ -163,176 +163,162 @@ These batches are wrapped up together in a ``MultiAgentBatch``,
- - The rollout workers have been already created ahead of time in the ``WorkerSet``, so the execution plan function is only defining a sequence of operations over the results of the rollouts.
+        # Update worker weights with the weights from the trained
+        # policy on the local worker.
+        self.workers.sync_weights()
 
-These iterators represent the infinite stream of data items that can be produced from the dataflow.
-Each operator (e.g., ``ConcatBatches``, ``TrainOneStep``), executes an operation over each item and returns a transformed item (e.g., concatenated batches, learner stats from training). Finally, some operators such as TrainOneStep have the *side-effect* of updating the rollout worker weights (that's why ``TrainOneStep`` takes the list of worker actors ``workers`` as an argument).
-
-Understanding and Debugging Execution Plans
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Execution plans are based on Ray `parallel iterators <https://github.com/ray-project/ray/blob/master/python/ray/util/iter.py>`__ and can be inspected similarly. For example, suppose you wanted to print out the intermediate data items during training. This can be done by inserting a print function into the dataflow, e.g., for A2C:
-
-.. code-block:: python
+        return train_results
 
-    def debug_print(item):
-        print("I saw", type(item))
-        return item
+As you can see each call to ``training_iteration`` does one sampling and training update. A dictionary is a returned that contains the results of the training update.
+ Its generally recommended that the dictionary map keys of type ``str`` to values that are of type ``float`` or to dictionaries of the same form, allowing for a nested structure.
 
-    train_op = rollouts \
-        .combine(ConcatBatches(
-            min_batch_size=config["train_batch_size"])) \
-        .for_each(debug_print) \
-        .for_each(TrainOneStep(workers))
-
-You'll see output like this on the console:
-
-.. code-block:: bash
+Training Iteration Function Utilities
+~~~~~~~~~~~~~~~~~~~~~~~
 
-    (pid=6555) I saw <class 'ray.rllib.policy.sample_batch.SampleBatch'>
-    (pid=6555) I saw <class 'ray.rllib.policy.sample_batch.SampleBatch'>
-    (pid=6555) I saw <class 'ray.rllib.policy.sample_batch.SampleBatch'>
-    (pid=6555) I saw <class 'ray.rllib.policy.sample_batch.SampleBatch'>
+RLlib provides a collection of utilities that abstract away common tasks in RL training.
 
-It is important to understand that the iterators of an execution plan are evaluated *lazily*. This means that no computation happens until the `trainer <#trainers>`__ tries to read the next item from the iterator (i.e., get the next training result for a ``Trainer.train()`` call).
+**Sample Batches** (`sample_batch.py <https://github.com/ray-project/ray/blob/master/rllib/policy/sample_batch.py>`__):
```

Comment:
This is also nice, but I think our standard class reference is not a github link but the class documentation inside our own docs. :py:class:`~ray.rllib.policy. sample_batch.SampleBatch`

Reply:
I can do this, but this doesn't create a link to documentation inside of our docs AFAICT.

I'm going to try linking directly to the API page in our docs.